### PR TITLE
Update the design of notifications

### PR DESCRIPTION
**Pull request by janfaracik**
This PR updates the design of notifications in Jenkins. This PR makes them a little more playful and vibrant, and it also brings them inline with the rest of the Jenkins interface.

**Before**
image

**After**

https://user-images.githubusercontent.com/43062514/188011724-d159584a-a505-4287-9274-2f99f8fdd275.mov

image
image
image
image

The examples above were produced by the Design Library, see https://github.com/jenkinsci/design-library-plugin/pull/147

### Proposed changelog entries

- Update the design of notifications

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7049"></a>

### CODE REVIEW

File: war/src/main/less/modules/notifications.less
Comment (NotMyFault) [inline war/src/main/less/modules/notifications.less:4]: The pop-up may get lost in the bottom left corner, I'd say, depending on the view you are at.

Did you try to keep the center position with the new animation?

**Comment by daniel-beck** [comment]
Does this implement https://github.com/jenkinsci/jenkins/pull/6408#discussion_r944483750 or how are they related?

**Comment by janfaracik** [comment]
> Does this implement [#6408 (comment)](https://github.com/jenkinsci/jenkins/pull/6408#discussion_r944483750) or how are they related?

The two PRs are completely independent of one another 👍 No shared dependencies etc.

**Comment by daniel-beck** [comment]
@janfaracik To clarify, the other PR changes the "Build scheduled" popup after I click "Build Now", while this PR take care of the message shown across the top after I select "Apply" on a form?

**Review by timja — COMMENTED** [review]
seems to be transparent? at least on the default one tested via design library.

the apply button looks really nice though.

image

File: war/src/main/less/modules/notifications.less
Comment (timja) [reply war/src/main/less/modules/notifications.less:4]: it looks nice where it currently is to me

**Comment by daniel-beck** [comment]
> @janfaracik To clarify, the other PR changes the "Build scheduled" popup after I click "Build Now", while this PR take care of the message shown across the top after I select "Apply" on a form?

Looks like it:

Screenshot 2022-09-05 at 17 35 58

Although the position looks kinda odd if the form is short, not where I would have expected it.

Screenshot 2022-09-05 at 17 36 29

File: war/src/main/less/modules/notifications.less
Comment (timja) [reply war/src/main/less/modules/notifications.less:4]: could possibly be in the top right?

Comment (janfaracik) [reply war/src/main/less/modules/notifications.less:4]: image

Screenshot 2022-09-05 at 23 27 47

Here it is in the top center/right. I'm not too sure how I feel about it up there, it all feels quite busy at the top and it might prevent a user from clicking a control/nav bar.

How would the bottom right sound? It's probably the emptiest corner in Jenkins.

Comment (timja) [reply war/src/main/less/modules/notifications.less:4]: Sure let’s try that

Comment (janfaracik) [reply war/src/main/less/modules/notifications.less:4]: image

Here it is in the bottom right, I definitely prefer that to the top imo - it's a lot less busy. 

What are your thoughts guys?

Comment (timja) [reply war/src/main/less/modules/notifications.less:4]: what if it's in the top but always below the header?

**Comment by NotMyFault** [comment]
>   Although the position looks kinda odd if the form is short, not where I would have expected it.

The same is the case if a view has no sidebar, the notification bar overlaps the buttons:

Screenshot 2022-09-14 at 23 47 48

I'm still in favor of leaving the old position as is and have it at the upper top part of the page. That's the least location notification bars can overlap content, I guess 🤔

File: war/src/main/less/modules/notifications.less
Comment (NotMyFault) [reply war/src/main/less/modules/notifications.less:4]: Bottom left, next to save and apply buttons, sounds less noisy to me.

Comment (janfaracik) [reply war/src/main/less/modules/notifications.less:4]: > what if it's in the top but always below the header?

I think that might introduce complexities, e.g. having to account for the visible portion of the header if the user is slightly scrolled down the page.

> Bottom left, next to save and apply buttons, sounds less noisy to me.

Are you both okay with bottom left then? We can always tweak it in future once page layout changes start coming in if needs be.

**Comment by timja** [comment]
@jenkinsci/core-security-review is there a reason this has been waiting a month?

I don't believe agreeing where on the page this goes impacts the security review?

**Review by yaroslavafenkin — COMMENTED** [review]
Looked through the code, no triggers for me.

**Review by timja — APPROVED** [review]
LGTM

**Review by NotMyFault — APPROVED** [review]
🎉

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
